src/lib/containers/entity_finder/tree/TreeView.tsx

```diff
@@ -0,0 +1,253 @@
+
+  return (
+    <div className="EntityFinderTreeView" style={{ height: '500px' }}>
+      {/* <div className={`EntityFinderTreeView__SelectionHeader`}></div> */}
+      <div style={{ overflow: 'auto' }}>
+        {isLoading ? (
+          <div className="spinner" />
+        ) : (
+          <>
+            {showFakeRootNode && (
+              <div
+                style={{ paddingLeft: `5px` }}
+                className={`EntityFinderTreeView__Row${
+                  currentContainer === 'root'
+                    ? ' EntityFinderTreeView__Row__Selected'
+                    : ''
+                }`}
+                onClick={() => {
+                  setCurrentContainer('root')
+                }}
+              >
+                <div
+                  className={'EntityFinderTreeView__Row__ExpandButton'}
+                  onClick={e => {
+                    e.stopPropagation()
+                    setExpandFakeRoot(!expandFakeRoot)
+                  }}
+                >
+                  {expandFakeRoot ? '▾' : '▸'}
+                </div>
+                <span></span>{' '}
+                <div
+                  style={{ width: 'min-content' }}
+                  onClick={e => e.stopPropagation()}
+                >
+                  <Dropdown
+                    style={{
+                      position: 'static',
+                    }}
+                  >
+                    <Dropdown.Toggle
+                      variant="light-primary-500"
+                      id="dropdown-basic"
+                    >
+                      {scope}
+                    </Dropdown.Toggle>
+                    <Dropdown.Menu>
+                      {Object.values(FinderScope).map(s => {
+                        return (
+                          <Dropdown.Item
+                            key={s}
+                            onClick={e => {
+                              console.log('setting scope', s)
+                              e.stopPropagation()
+                              setScope(s)
+                            }}
+                          >
+                            {s}
+                          </Dropdown.Item>
+                        )
+                      })}
+                    </Dropdown.Menu>
+                  </Dropdown>
+                </div>
+              </div>
+            )}
+            <div style={!expandFakeRoot ? { display: 'none' } : {}}>
+              {topLevelEntities?.map((entity: EntityHeader | ProjectHeader) => {
+                return (
+                  <TreeViewRow
+                    key={entity.id}
+                    sessionToken={sessionToken}
+                    entityHeader={entity}
+                    selectedId={currentContainer}
+                    setSelectedId={(entityId: string) => {
+                      setCurrentContainer(entityId)
+                    }}
+                    autoExpand={entityId => {
+                      return !!(
```

Comment:
is this a double logical negation or something else?  and why?

Reply:
`initialContainerPath` may be undefined. To improve clarity, I will move the `!!` to `initialContainerPath` rather than the entire expression